test/oli_web/live/payments_live_test.exs

```diff
@@ -0,0 +1,106 @@
+defmodule OliWeb.PaymentsLiveTest do
+  use ExUnit.Case
+  use OliWeb.ConnCase
+
+  import Phoenix.LiveViewTest
+  import Oli.Factory
+
+  alias OliWeb.Router.Helpers, as: Routes
+  alias Oli.Delivery.Paywall.Payment
+
+  defp create_product(_conn) do
+    product =
+      insert(:section, type: :blueprint, requires_payment: true, amount: Money.new(:USD, 10))
+
+    [product: product]
+  end
+
+  defp live_view_payments_route(product_slug) do
+    Routes.live_path(OliWeb.Endpoint, OliWeb.Products.PaymentsView, product_slug)
+  end
+
+  @live_view_payment_route Routes.live_path(
+                             OliWeb.Endpoint,
+                             OliWeb.Products.PaymentsView,
+                             "test_product"
+                           )
+
+  @live_view_product_route Routes.live_path(
+                             OliWeb.Endpoint,
+                             OliWeb.Products.ProductsView,
+                             "test_product"
+                           )
+
+  describe "user cannot access when is not logged in" do
+    test "redirects to new session when accessing the index view", %{conn: conn} do
+      {:error,
+       {:redirect,
+        %{
+          to:
+            "/authoring/session/new?request_path=%2Fauthoring%2Fproducts%2Ftest_product%2Fpayments"
+        }}} = live(conn, @live_view_payment_route)
+    end
+  end
+
+  describe "user cannot access when is logged in as an author but is not a system admin" do
```

Comment:
Are you sure this is correct? I believe the authors are able to access this view, but just the authors associated to the project from where the product was created

Reply:
Good. Maybe I did it without knowing well how the roles and permissions of the different users work inside torus. I will rethink it then!